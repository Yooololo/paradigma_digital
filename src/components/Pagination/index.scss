.pagination {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: $padding;

  p {
    margin-top: $margin;
    background-color: $blue;
    padding: $margin;
    border-radius: $box-border-radius * 2;
    color: $white;
  }
  &.dark {
    p {
      border: 1px solid $white;
    }
  }
  &.light {
    p {
      border: 1px solid $grey;
    }
  }
  &__options {
    display: flex;
    list-style-type: none;
    align-items: center;

    &__button {
      cursor: pointer;
      font-weight: 500;
      margin: 0 $margin;
      padding: $margin;
      background-color: $blue;
      color: $white;
      font-size: $font-size;
      border-radius: $box-border-radius * 2;
      border: 1px solid $white;
    }

    &__page {
      cursor: pointer;
      padding: calc($margin/2);
      margin: 0 calc($margin/2);
      font-size: $font-size;
      &.dark {
        color: $white;
      }
      &.light {
        color: $black;
      }
      &:hover {
        color: $blue;
        font-weight: 600;
      }
      &__active {
        background-color: $blue-50;
        color: $white !important;
        border-radius: $box-border-radius;
        border: 1px solid $grey-dark;
        &.dark {
          border: 1px solid $white;
        }
      }
    }
  }
  .disabled {
    cursor: default;
    background-color: $grey;
  }
}

.hidden {
  height: calc(1.5 * $margin + 2 * $padding);
}
