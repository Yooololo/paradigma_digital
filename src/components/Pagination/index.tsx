import { useContext } from "react";
import { FormattedMessage } from "react-intl";
import { ThemeColorContext } from "../../context/ThemeColorWrapper";

const Pagination = ({
  podcastsPerPage,
  totalPodcasts,
  pagination,
  previousPage,
  nextPage,
  currentPage,
}: {
  podcastsPerPage: number;
  totalPodcasts: number;
  pagination: (pageNumber: number) => void;
  previousPage: () => void;
  nextPage: () => void;
  currentPage: number;
}) => {
  const { themeColor } = useContext(ThemeColorContext);
  const pageNumbers = [];

  for (let i = 1; i <= Math.ceil(totalPodcasts / podcastsPerPage); i++) {
    pageNumbers.push(i);
  }

  return (
    <>
      {pageNumbers.length > 1 ? (
        <div className={`pagination ${themeColor}`}>
          <ul className="pagination__options">
            <li
              onClick={previousPage}
              className={`pagination__options__button ${
                currentPage === 1 ? "disabled" : ""
              }`}
            >
              <FormattedMessage id="app.prev" />
            </li>
            {pageNumbers.map((number) => (
              <li
                key={number}
                onClick={() => {
                  pagination(number);
                }}
                className={`pagination__options__page ${themeColor} ${
                  currentPage === number
                    ? "pagination__options__page__active"
                    : ""
                }`}
              >
                {number}
              </li>
            ))}
            <li
              onClick={nextPage}
              className={`pagination__options__button ${
                currentPage === Math.ceil(totalPodcasts / podcastsPerPage)
                  ? "disabled"
                  : ""
              }`}
            >
              <FormattedMessage id="app.next" />
            </li>
          </ul>
          <p>
            {podcastsPerPage * currentPage - podcastsPerPage + 1} -{" "}
            {podcastsPerPage * currentPage < totalPodcasts
              ? podcastsPerPage * currentPage
              : totalPodcasts}
          </p>
        </div>
      ) : (
        <div className="hidden" />
      )}
    </>
  );
};
export default Pagination;
