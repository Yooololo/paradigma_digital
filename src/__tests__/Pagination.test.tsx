import Pagination from "../components/Pagination";
import React from "react";
import { cleanup, render } from "@testing-library/react";
import { IntlProvider } from "react-intl";
import English from "../languages/en.json";

afterEach(cleanup);

test("Pagination should render the pagination component correctly", () => {
  const mockedThemeColor = "dark";
  const mockUseContext = jest.fn();
  mockUseContext.mockReturnValueOnce(mockedThemeColor);
  jest
    .spyOn(React, "useContext")
    .mockImplementationOnce(() => mockUseContext());

  const mockedPodcastsPerPage = 10;
  const mockedTotalPodcasts = 50;
  const mockedPagination = jest.fn();
  const mockedPreviousPage = jest.fn();
  const mockedNextPage = jest.fn();
  const mockedCurrentPage = 1;

  const { container } = render(
    <IntlProvider locale="en-US" messages={English}>
      <Pagination
        podcastsPerPage={mockedPodcastsPerPage}
        totalPodcasts={mockedTotalPodcasts}
        pagination={mockedPagination}
        previousPage={mockedPreviousPage}
        nextPage={mockedNextPage}
        currentPage={mockedCurrentPage}
      />
    </IntlProvider>
  );

  expect(container).toMatchSnapshot();
  expect(mockUseContext).toHaveBeenCalled();
  expect(container.querySelector(".pagination")).toBeTruthy();
  expect(container.querySelector(".pagination__options")).toBeTruthy();
  expect(container.querySelector(".pagination__options__button")).toBeTruthy();
  expect(container.querySelectorAll(".pagination__options__page")).toHaveLength(
    5
  );
});
