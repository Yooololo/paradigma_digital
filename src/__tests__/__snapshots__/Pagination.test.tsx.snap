// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render the pagination component correctly 1`] = `
<div>
  <div
    class="pagination undefined"
  >
    <ul
      class="pagination__options"
    >
      <li
        class="pagination__options__button disabled"
      >
        Prev
      </li>
      <li
        class="pagination__options__page undefined pagination__options__page__active"
      >
        1
      </li>
      <li
        class="pagination__options__page undefined "
      >
        2
      </li>
      <li
        class="pagination__options__page undefined "
      >
        3
      </li>
      <li
        class="pagination__options__page undefined "
      >
        4
      </li>
      <li
        class="pagination__options__page undefined "
      >
        5
      </li>
      <li
        class="pagination__options__button "
      >
        Next
      </li>
    </ul>
    <p>
      1
       -
       
      10
    </p>
  </div>
</div>
`;
